.github/workflows/ci-test-hosted.yml

```
@@ -252,6 +258,17 @@ jobs:
           working-directory: app/client
           env: "NODE_ENV=development"
 
+      - name: Stop Monitoring
+        run: |
+          kill $(cat resource_monitor_pid) || true
+          echo "Resource monitoring stopped."
+  
+      - name: Upload Resource Usage Log
+        uses: actions/upload-artifact@v3
```

actionlint (1.7.4)

the runner of "actions/upload-artifact@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)